pytorch_lightning/core/lightning.py

```diff
@@ -449,7 +449,7 @@ def all_gather(
 
         data = convert_to_tensors(data, device=self.device)
         all_gather = partial(all_gather, group=group, sync_grads=sync_grads)
-        return apply_to_collection(data, torch.Tensor, all_gather)
+        return apply_to_collection(data, all_gather, dtype=torch.Tensor)
```

Comment:
Any reasons for switching them ? It is breaking BC.

Reply:
Can't have keyword arguments before positional

Reply:
then just call dem as `argument=value`

Reply:
The issue is not the order here but in the signature